gammapy/maps/base.py

```diff
@@ -1032,3 +1033,56 @@ def __repr__(self):
         str_ += "\tunit  : {!r} \n".format(str(self.unit))
         str_ += "\tdtype : {} \n".format(self.data.dtype)
         return str_
+
+    def __add__(self, other):
+        """ Add two maps with compatible geometries and units together
```

Comment:
See my PR with the alternative implementation.
I suggest to not put docstrings, or to spend effort to make them precise.
The current "add two maps" isn't really correct, the method then also handles other inputs.

Reply:
OK. Will be improved in arithmetic PR